### PR TITLE
Add NI-DCPower proto file and update generated `nidcpower` files

### PR DESCRIPTION
**Pull request by tdunkle**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~

This change does not need to be customer-facing.

~- [ ] I've added tests applicable for this pull request~

Showing that the existing NI-DCPower system-level tests pass is sufficient.

### What does this Pull Request accomplish?

This pull request adds the NI-DCPower proto file to nimi-python, so that the NI-DCPower python API has a gRPC option enabled.

### List issues fixed by this Pull Request below, if any.

* [Internal NI bug AB#2149854](https://ni.visualstudio.com/DevCentral/_workitems/edit/2149854/)

### What testing has been done?

Installed python 3.7.9 and synced the nimi-python repo to a VM that has NI-DCPower 2022 Q4 installed. Observed that I could build nidcpower with `tox -e py37-codegen -- nidcpower`, and that I could run the system level tests with `tox -c generated\nidcpower\tox-system_tests.ini`, and all but one passed initially. I had to make a change to fix `test_create_and_delete_advanced_sequence_bad_type()`. Test output with all passing, for reference: [dcpower_test_output.txt](https://github.com/ni/nimi-python/files/9864065/dcpower_test_output.txt)

### CODE REVIEW

File: src/nidcpower/system_tests/test_system_nidcpower.py
Comment (tdunkle) [inline src/nidcpower/system_tests/test_system_nidcpower.py:529]: @DavidCurtiss This was the only change I had to make in the system tests to get them all to pass for nidcpower. The error I get with the original attribute is: `nidcpower.errors.DriverError: -1074118544: The specified attribute is not supported in an advanced sequence for this device.` I'm not sure why it doesn't work anymore, but it seems like which attribute is used doesn't matter for this test?

Comment (DavidCurtiss) [reply src/nidcpower/system_tests/test_system_nidcpower.py:529]: Does that error happen only over gRPC, or via the local tests, too?

**Comment by dmondrik** [comment]
I'd like to see [my PR for gRPC session options](https://github.com/ni/nimi-python/pull/1835) to go in first.

File: src/nidcpower/system_tests/test_system_nidcpower.py
Comment (marcoskirsch) [reply src/nidcpower/system_tests/test_system_nidcpower.py:529]: This is fishy, need to get to the bottom of it.

Comment (ni-jfitzger) [reply src/nidcpower/system_tests/test_system_nidcpower.py:529]: See #1839 for what I believe is the proper fix.

Comment (dmondrik) [reply src/nidcpower/system_tests/test_system_nidcpower.py:529]: @ni-jfitzger what is keeping that fix (1839, test removal) from going in?

Comment (ni-jfitzger) [reply src/nidcpower/system_tests/test_system_nidcpower.py:529]: We're just waiting for @marcoskirsch to review and confirm that our conclusions are reasonable.

**Comment by marcoskirsch** [comment]
Travis CI PR checks are failing. Specifically a check we added to ensure codegen is up to date (contributors forget to fix it).
See: https://app.travis-ci.com/github/ni/nimi-python/jobs/587164615

Please fix it, I wonder if some of the gRPC changes made it so it's always broken.

**Comment by DavidCurtiss** [comment]
> Please fix it, I wonder if some of the gRPC changes made it so it's always broken.

@marcoskirsch, no it's not always broken. It's just that Dan's change going in required all of the other PRs to update. And each one is dealing with different issues now. (My DMM PR, for example, is dealing with other hapigen changes that have gone in. This DCPower one, I'm not sure if there's any problems, but Tom is focusing on FGEN right now.)

**Comment by marcoskirsch** [comment]
The failure is:

> The command "python tools/ensure_codegen_up_to_date.py" failed and exited with 1 during .

Is the failure inaccurate?

Regardless, there's a conflict to be resolved here blocking the merge.

**Comment by marcoskirsch** [comment]
The failure is:

> The command "python tools/ensure_codegen_up_to_date.py" failed and exited with 1 during .

Is the failure inaccurate?

Regardless, there's a conflict to be resolved here blocking the merge.

**Comment by marcoskirsch** [comment]
The failure is:

> The command "python tools/ensure_codegen_up_to_date.py" failed and exited with 1 during .

Is the failure inaccurate?

Regardless, there's a conflict to be resolved here blocking the merge.

**Comment by DavidCurtiss** [comment]
When I said "I'm not sure if there's any problems" with NI-DCPower, I meant besides needing to regen after the merge. We'll have to see when Tom finishes with NI-FGEN. Whereas with other drivers, like NI-DMM, there are additional issues cropping up that we're fighting off, preventing us from posting timely updates to some of these build/test failures.

But there's nothing specific to the gRPC changes that has caused the codegen up-to-date errors. It's just developers running into the same problem that caused the up-to-date check to be added in the first place.

**Comment by tdunkle** [comment]
#1846 is a prerequisite to this PR.

File: src/nidcpower/custom_types/lcr_measurement.py
Comment (dmondrik) [inline src/nidcpower/custom_types/lcr_measurement.py:145]: The `if` and `else` clauses look identical to me. What's going on here?

File: src/nidcpower/metadata/functions.py
Comment (dmondrik) [inline src/nidcpower/metadata/functions.py:245]: Array of addresses is wrong. Something didn't get set correctly.

File: src/nidcpower/system_tests/test_system_nidcpower.py
Comment (dmondrik) [inline src/nidcpower/system_tests/test_system_nidcpower.py:1025]: This comment doesn't make sense to me for `reset()`. Was it copy/paste from one of the tests below?

**Review by dmondrik — CHANGES_REQUESTED** [review]
The array-of-address seems the biggest issue. For other APIs it was solved in the hapigen metadata. Let me know if you want me to scrub that with you.

File: src/nidcpower/custom_types/lcr_measurement.py
Comment (DavidCurtiss) [reply src/nidcpower/custom_types/lcr_measurement.py:145]: One is `ac_voltage_real`; the other is `ac_voltage.real` (with a `.` before `real` instead of a `_`).

In the proto, `ac_voltage` is a `message` with `real` and `imaginary` members.

File: src/nidcpower/metadata/functions.py
Comment (DavidCurtiss) [reply src/nidcpower/metadata/functions.py:245]: This is a continued result of the current state of `address`/`ViAddr` parameters. The same thing can be seen [here](https://github.com/ni/nimi-python/blob/master/src/nidmm/metadata/functions.py#L1336), for example.

File: generated/nidcpower/nidcpower/_grpc_stub_interpreter.py
Comment (DavidCurtiss) [inline generated/nidcpower/nidcpower/_grpc_stub_interpreter.py:9]: Based on the TravisCI errors, it looks like we're going to need to add the `# noqa: F401` comment here (specifically, in the .mako).

File: src/nidcpower/system_tests/test_system_nidcpower.py
Comment (tdunkle) [reply src/nidcpower/system_tests/test_system_nidcpower.py:1025]: No, I added the comment. Internal NI bug 2209572 captures the work of fixing this test (and a few others in this file), though I'm not sure that we want to refer to that item in a comment since folks outside of NI can't access it.

It's been difficult (read: nearly impossible) to get gRPC tests enabled in one PR with 100% of them passing quickly enough for the change to be submitted without being broken by anything upstream. So, the intent here is to get the current PR submitted with minimal tests disabled for gRPC, then immediately follow it up with other change(s) fixing and re-enabling these tests.

File: generated/nidcpower/nidcpower/_grpc_stub_interpreter.py
Comment (tdunkle) [reply generated/nidcpower/nidcpower/_grpc_stub_interpreter.py:9]: Done!

File: src/nidcpower/metadata/functions.py
Comment (dmondrik) [reply src/nidcpower/metadata/functions.py:245]: Ah, I see what you're saying David. If the generated python code is correct despite the metadata being incorrect, that's not something to solve here in this PR.

**Review by dmondrik — APPROVED** [review]
LGTM

File: src/nidcpower/system_tests/test_system_nidcpower.py
Comment (ni-jfitzger) [inline src/nidcpower/system_tests/test_system_nidcpower.py:1107]: Is this part intentionally different from the content of `test_system_nidmm.py`?

Comment (DavidCurtiss) [reply src/nidcpower/system_tests/test_system_nidcpower.py:1107]: To add some more context to what Tom said: In most drivers so far, for test functions that construct a Session, we have one version of the test in TestLibrary, and another in TestGrpc, where the only difference is what additional kwargs we pass to the Session constructor. Rather than duplicate such a large number of tests here, we decided to use this session_creation_kwargs fixture to supply the different kwargs per class.

For consistency, it might be a good idea to modify the system tests for all drivers to use this session_creation_kwargs style, even though the other system test files might only use it in one or two functions. But that at least doesn't need to be done in this PR, IMO.

File: src/nidcpower/custom_types/lcr_load_compensation_spot.py
Comment (tdunkle) [inline src/nidcpower/custom_types/lcr_load_compensation_spot.py:77]: Need to pull this out of the current PR.

File: generated/nidcpower/nidcpower/_grpc_stub_interpreter.py
Comment (tdunkle) [inline generated/nidcpower/nidcpower/_grpc_stub_interpreter.py:84]: Need a follow-on work item to inspect the not implemented functions.

Comment (tdunkle) [reply generated/nidcpower/nidcpower/_grpc_stub_interpreter.py:84]: Created internal NI bug 2210662.

**Review by ni-jfitzger — APPROVED** [review]
Looks good. I'll merge after the tests complete.

**Comment by ni-jfitzger** [comment]
FYI @marcoskirsch merged to unblock upcoming work that will rely on this change.

**Pull request merged**
